src/Infrastructure.VS/Roslyn/SolutionRoslynAnalyzerManager.cs

```diff
@@ -115,38 +121,11 @@ public void Dispose()
         {
             return;
         }
-        connectedModeAnalyzerProvider.AnalyzerUpdatedForConnection -= HandleConnectedModeAnalyzerUpdate;
+        activeConfigScopeTracker.CurrentConfigurationScopeChanged -= OnConfigurationScopeChanged;
+        activeSolutionTracker.ActiveSolutionChanged -= OnActiveSolutionChanged;
         disposed = true;
     }
 
-    internal /* for testing */ async Task HandleConnectedModeAnalyzerUpdateAsync(AnalyzerUpdatedForConnectionEventArgs args)
-    {
-        var analyzersToUse = await ChooseAnalyzersAsync();
-        lock (lockObject)
-        {
-            ThrowIfDisposed();
-            UpdateAnalyzersIfChanged(analyzersToUse);
-        }
-    }
-
-    private void HandleConnectedModeAnalyzerUpdate(object sender, AnalyzerUpdatedForConnectionEventArgs args)
-    {
-        HandleConnectedModeAnalyzerUpdateAsync(args).Forget();
-    }
-
-    private void UpdateCurrentSolutionInfo(string solutionName, out bool isSameSolution)
-    {
-        if (currentState is { SolutionName: var currentSolutionName } && solutionName == currentSolutionName)
-        {
-            isSameSolution = true;
-        }
-        else
-        {
-            currentState = new SolutionStateInfo(solutionName);
-            isSameSolution = false;
-        }
-    }
-
     private async Task<ImmutableArray<AnalyzerFileReference>> ChooseAnalyzersAsync() =>
         ChooseAnalyzers(await connectedModeAnalyzerProvider.GetOrNullAsync());
```

Comment:
We should provide the config scope here

Reply:
I don't want to change the interface IconnectedModeAnalyzerProvider in this PR. I added a comment in this task: https://sonarsource.atlassian.net/browse/SLVS-1518